[core] Expose `dbPromise` in `StateStore`

confused-Techie opened the pull request:
The getter method of `dbPromise` within `StateStore` is relied on by the community package [`project-plus`](https://web.pulsar-edit.dev/packages/project-plus), but with the release of Pulsar `v1.122.0` and specifically the introduction of the new StateStore database in #917 that method has been hidden away within each database's own code. 

But because both database's still have this method, it seems entirely possible we could just expose this the same way we expose their other methods.

Closes #1169

savetheclocktower reviewed (changes requested):
Looks great, but can we add a quick spec to `state-store-spec.js`? If this is going to be (informally) part of the contract, we should protect against future regressions.

confused-Techie commented:
@savetheclocktower Thanks for giving this a review, I've added the specs and that lead to me finding out this was still somewhat broken, so I've also gone ahead and fixed it. We should be good to go now

confused-Techie commented:
Just wanting to add it looks like the only failure in CI is due to the issues resolved in #1173 so they shouldn't be considered here on this PR.

confused-Techie commented:
Thanks @mauricioszabo
